### PR TITLE
:art: refactorting getuserpublickey and restore bug

### PR DESCRIPTION
**Pull request by pedrobertao**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -286,14 +287,18 @@ class App extends Component {
...
 
   componentDidMount() {
-    this._getFreeze()
     this._getPrice()
-    this._getPublicKey()
     this._setNodes()
+    this._loadUserData()
+  }
+
+  _loadUserData = async () => {
```
Comment (julianorafael) [inline App.js]: shouldn't be async. Isn't returning a promise or waiting for one in the body of the function.

File: App.js [hunk 2]
```diff
@@ -332,8 +337,9 @@ class App extends Component {
   }
 
   render() {
+    const contextProps = { ...this.state, restoreWalletData: this._loadUserData }
```
Comment (julianorafael) [inline App.js]: why can't you reference the method on the state, like `getPublicKey`?

File: src/scenes/Freeze/index.js [hunk 3]
```diff
@@ -24,13 +24,11 @@ class FreezeScene extends Component {
   }
 
   componentDidMount() {
-    this._navListener = this.props.navigation.addListener('didFocus', () => {
-      this.loadData()
-    })
+    this.didFocusSubscription = this.props.navigation.addListener('didFocus', this.loadData)
```
Comment (julianorafael) [inline src/scenes/Freeze/index.js]: if it's not a React lifecycle method, start his name with underscore.

File: src/scenes/Login/index.js [hunk 4]
```diff
@@ -74,9 +74,9 @@ class LoginScene extends Component {
         )
       } else {
         try {
-          const userPublicKey = await Client.getPublicKey()
+          const userPublicKey = await getUserPublicKey()
           const { address } = await getUserSecrets()
-          if (userPublicKey !== address) this.createKeyPair();
+          if (userPublicKey !== address) await this.createKeyPair();
```
Comment (julianorafael) [inline src/scenes/Login/index.js]: why do you need to await? You can trigger and forget, you are not going to return any value whatsoever.

File: App.js (hunk 2 above)
Comment (pedrobertao) [reply App.js]: It is undefined this way. The function doesn't exist during with the state constructor.

File: src/scenes/Login/index.js (hunk 4 above)
Comment (pedrobertao) [reply src/scenes/Login/index.js]: Well, if you don't wait to createkeypair it is going to navigate home while the user might have or not a public key

File: App.js [hunk 5]
```diff
@@ -336,8 +339,15 @@ class App extends Component {
   }
 
   render() {
+    const contextProps = {
+      ...this.state,
+      restoreWalletData: this._loadUserData,
+      getFreeze: this._getFreeze,
```
[the remark this answers is not in the document]
Comment (pedrobertao) [reply App.js]: Next pr i will put this

**Pull request merged**